### PR TITLE
LPS-159320 | Add supported query params to Headless Admin Taxonomy API Explorer

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add a test to assert to support query params to Headless Delivery API Explorer

**Test Plan**
Given a Taxonomy Vocabulary with assetLibrary
When with curl I request getAssetLibraryTaxonomyVocabulariesPage with restrictFields=actions,assetLibraryKey,assetTypes,availableLanguages,creator,dateCreated,dateModified,
description,externalReferenceCode,id,numberOfTaxonomyCategories
Then in a response I receive a with name field values only

Given a Taxonomy Vocabulary
When with curl I request getSiteTaxonomyVocabulariesPage with aggregationTerms=id
Then in a response I receive numberOfOccurrences and term with correct value in facets fields

Given a AssetLibrary with keyword created
When with curl I request getAssetLibraryKeywordsPage with fields=name
Then in a response I receive a body with name values only

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-159320

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro [hunk 1]
```diff
@@ -1,5 +1,31 @@
 definition {
 
+	macro _curlTaxonomyKeyWords {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro:3]: typo, `_curlTaxonomyKeywords`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/HeadlessAdminTaxonomySupportDifferentParameters.testcase [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Headless Frontend Infrastructure";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		TaxonomyVocabularyAPI.setUpGlobalSiteId();
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/HeadlessAdminTaxonomySupportDifferentParameters.testcase:11]: I'm not sure whether this approach is safe... This way, if anyone needs to use our macro will have to set up the siteId themselves, which may be a bit problematic, maybe it is better if the parts of asking for siteId stays inside the macro. @yanan @Hijky what do you think about it?

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro [hunk 3]
```diff
@@ -112,6 +184,28 @@ definition {
...
+			siteId = "${siteId}");
+
+		var response = JSONCurlUtil.get("${curl}");
+
+		return "${response}";
+	}
+
+	macro setUpGlobalSiteId {
+		var siteId = JSONGroupAPI._getGroupIdByNameNoSelenium(
+			groupName = "Guest",
```
Comment (yanan) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro:201]: Just a little curious that the name is "setUpGlobalsiteId" but we're using "Guest" site not using "Global" site here.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/HeadlessAdminTaxonomySupportDifferentParameters.testcase (hunk 2 above)
Comment (yanan) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/HeadlessAdminTaxonomySupportDifferentParameters.testcase:11]: Agree. What about we pass the siteId as a parameter inside of method in macro? In the codes now we are using "Guest" site, if you'd like to use "Global" site then we can add one more parameter "groupName".

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro [hunk 4]
```diff
@@ -29,6 +59,16 @@ definition {
 		return "${curl}";
 	}
 
+	macro assertInFacetsWithCorrectValue {
+		Variables.assertDefined(parameterList = "${expectedValue},${taxonomyVocabularyId}");
+
+		var actualValue = JSONUtil.getWithJSONPath("${responseToParse}", "$.facets..facetValues[?(@.term=='${taxonomyVocabularyId}' && @.numberOfOccurrences==${expectedValue})].numberOfOccurrences");
```
Comment (yanan) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro:65]: Good to learn the jsonpath part!

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by Hijky** [comment]
@magjed4289 @yanan I edit the typo and I modify the function for _curlTaxonomyVocabulary, you can check it again.

**Review by magjed4289 — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**